packages/babel-plugin-proposal-decorators/test/fixtures/2022-03-classes--to-es2015/replacement-static-installed-on-correct-class/output.js

```diff
@@ -8,8 +8,8 @@ new (_x = /*#__PURE__*/new WeakMap(), _m = /*#__PURE__*/new WeakSet(), (_temp =
       writable: true,
       value: void 0
     }), babelHelpers.defineProperty(this, "x", void 0)), (() => {
-      hasX = o => _x.has(o);
-      hasM = o => _m.has(o);
+      hasX = o => _x.has(babelHelpers.checkInRHS(o));
```

Comment:
Could we move the `.has` call to the helper, and just transform `#x in o` to `babelHelpers.privateHas(_x, o)`?

Reply:
I lean to a new assumption `inRHSIsObject` to skip the check. Otherwise we have to come up with at least three helpers:

- staticPrivateHas for `static #p = #p in C`
- privateHas for `#p = #p in this`
- privateAsPropertiesHas for `#p = #p in this`

And we can implement the `checkInRHS` helper as 
```js
function checkInRHS(v) {
  0 in v
}
```
so that the engine will do the checking job for us.

Reply:
Actually, we can't because `0 in v` is observable when using proxies.